### PR TITLE
1548: Selenium: open browser code does not account for default values of arguments and 1591: Selenium fix for  togther as these tickets are interdependent

### PR DESCRIPTION
**Pull request by brettleehari**
Addressed, 1548 and 1591 together as they are interdependent.

### CODE REVIEW

File: warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py [hunk 1]
```diff
@@ -480,20 +480,23 @@ def _make_ff(self, webdriver_remote_url, desired_capabilites, profile_dir,
...
                 ff_capabilities = webdriver.DesiredCapabilities.FIREFOX
-                if ff_capabilities['marionette']:
+                try:
                     ff_capabilities['acceptInsecureCerts'] = True
                     ffbinary = FirefoxBinary(binary)
                     browser = webdriver.Firefox(firefox_binary=ffbinary,
                                                 firefox_profile=profile_dir,
                                                 executable_path=gecko_path)
-                else:
+                except:
```
Comment (navvrnd) [inline warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py]: why this nested try catch? why can't this be handled in the outer try/except? Please handle the specific exceptions expected.

File: warrior/Actions/SeleniumActions/browser_actions.py [hunk 2]
```diff
@@ -208,23 +212,25 @@ def browser_launch(self, system_name, browser_name="all", type="firefox",
...
-                if binary != "" and gecko_path != "":
+                if browser_details["gecko_path"] not in [None, False] and \
+                                browser_details["type"] == "firefox":
+                    ff_profile = self.browser_object.set_firefoxprofile(proxy_ip, proxy_port)
                     browser_inst = self.browser_object.open_browser(
                         browser_details["type"], webdriver_remote_url,
                         binary=binary, gecko_path=gecko_path,
                         profile_dir=ff_profile)
                 else:
-                    pNote("Please provide valid path for binary/geckodriver")
```
Comment (kaheichan) [inline warrior/Actions/SeleniumActions/browser_actions.py]: This is supposed to be a failure case, shouldn't even attempt to launch browser here if geckodriver is not found

**Comment by kaheichan** [comment]
Code not in ready-to-review state, will add assignee once it is ready

**Comment by kaheichan** [comment]
A New PR #333 has been created to fit the scope of all related ticket

File: warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py (hunk 1 above)
Comment (kaheichan) [inline warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py]: The try except block ignore the error and force a Firefox browser to spawn
without logging the potential error

File: warrior/Actions/SeleniumActions/browser_actions.py [hunk 3]
```diff
@@ -187,15 +187,19 @@ def browser_launch(self, system_name, browser_name="all", type="firefox",
...
-
-        webdriver_remote_url = ip if str(remote).strip().lower() == "yes"\
-            else False
-
-        system = xml_Utils.getElementWithTagAttribValueMatch(self.datafile,
-                                                             "system",
-                                                             "name",
+            remote = data_Utils.getSystemData(self.datafile, system_name, "remote")
+        if binary is None:
+            binary = data_Utils.getSystemData(self.datafile, system_name, "binary")
```
Comment (kaheichan) [inline warrior/Actions/SeleniumActions/browser_actions.py]: binary, gecko_path. proxy_ip and proxy_port aren't in system
It's inside browser level

**Pull request closed without merging**